query-algebrizer/src/clauses/fulltext.rs

```diff
@@ -0,0 +1,310 @@
+                    Some(ref in_value) => in_value.clone(),
+                    None => bail!(ErrorKind::UnboundVariable((*in_var.0).clone())),
+                }
+            },
+            FnArg::Constant(NonIntegerConstant::Text(s)) => {
+                TypedValue::typed_string(s.as_str())
+            },
+            _ => bail!(ErrorKind::InvalidArgument(where_fn.operator.clone(), "string".into(), 2)),
+        };
+
+        // TODO: should we build the FQA in ::Matches, preventing nonsense like matching on ::Rowid?
+        let constraint = ColumnConstraint::Matches(QualifiedAlias(fulltext_values_alias.clone(), Column::Fulltext(FulltextColumn::Text)),
+                                                   QueryValue::TypedValue(search));
+        self.wheres.add_intersection(constraint);
+
+        if let VariableOrPlaceholder::Variable(ref var) = bindings[0] {
+            if self.bound_value(&var).is_some() || self.input_variables.contains(&var) {
+                // It's OK for this to collide: the two must unify.
+                unimplemented!();
+            }
+            self.constrain_var_to_type(var.clone(), ValueType::Ref);
+
+            let entity_alias = QualifiedAlias(datoms_table_alias.clone(), Column::Fixed(DatomsColumn::Entity));
+            self.column_bindings.entry(var.clone()).or_insert(vec![]).push(entity_alias);
+        }
+
+        if let VariableOrPlaceholder::Variable(ref var) = bindings[1] {
+            if self.bound_value(&var).is_some() || self.input_variables.contains(&var) {
+                // It's OK for this to collide: the two must unify.
+                unimplemented!();
+            }
+            self.constrain_var_to_type(var.clone(), ValueType::String);
+
+            let value_alias = QualifiedAlias(fulltext_values_alias.clone(), Column::Fulltext(FulltextColumn::Text));
+            self.column_bindings.entry(var.clone()).or_insert(vec![]).push(value_alias);
+        }
+
+        if let VariableOrPlaceholder::Variable(ref var) = bindings[2] {
+            if self.bound_value(&var).is_some() || self.input_variables.contains(&var) {
+                // It's OK for this to collide: the two must unify.
+                unimplemented!();
+            }
+            self.constrain_var_to_type(var.clone(), ValueType::Ref);
+
+            let tx_alias = QualifiedAlias(datoms_table_alias.clone(), Column::Fixed(DatomsColumn::Tx));
+            self.column_bindings.entry(var.clone()).or_insert(vec![]).push(tx_alias);
+        }
+
+        if let VariableOrPlaceholder::Variable(ref var) = bindings[3] {
+            if self.bound_value(&var).is_some() || self.input_variables.contains(&var) {
+                // It's OK for this to collide: the two must unify.
+                unimplemented!();
+            }
+            self.constrain_var_to_type(var.clone(), ValueType::Double);
+
+            // Right now we don't support binding a column to a constant value.  Therefore, we
```

Comment:
You have now done the work for `ground` to produce exactly the constant value, all the way out of the projector.  Can you translate this to a `(ground ?score 0.0)` rather than injecting a computed table into the mix?